journals/apps/core/management/commands/create_site.py

```diff
@@ -0,0 +1,155 @@
+from django.contrib.auth.models import Group, Permission
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from journals.apps.journals.models import JournalIndexPage
+from journals.apps.core.models import SiteConfiguration
+from journals.apps.theming.models import SiteBranding
+
+from wagtail.wagtailcore.models import Page, Site, Collection, GroupPagePermission, GroupCollectionPermission
+
+PERMISSIONS = {
+    'Editors':{
+        'page_permissions': ['add', 'edit'],
+        'collection_permissions': [
+            'add_image', 'change_image',
+            'add_document', 'change_document'
+        ],
+        'core_permissions': ['access_admin']
+    },
+    'Moderators':{
+        'page_permissions': ['add', 'edit', 'publish'],
+        'collection_permissions': [
+            'add_image', 'change_image', 'delete_image',
+            'add_document', 'change_document', 'delete_document'
+        ],
+        'core_permissions': ['access_admin', 'change_sitebranding']
+    }
+}
+
+class Command(BaseCommand):
+    help = 'Creates a new site and all necessary additional configs'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--sitename', help='Name of new site') # MITxPRO or edX
+        parser.add_argument('--hostname', help='Hostname of new site (e.g. journals.example.com)')
+        parser.add_argument('--port', help='Webserver port to listen to')
+
+
+    def create_index_page(self, sitename):
+        root_page = Page.get_root_nodes()[0]
+        index_page = JournalIndexPage(
+            title="{} Index Page".format(sitename),
+            intro="{} introduction".format(sitename)
+        )
+        root_page.add_child(instance=index_page)
```

Comment:
question: should we check if an index page for this site exists already and throw an error? What currently happens if it exists

Reply:
Since pages are not unique on title or intro, this will just create another index page with the same attributes.